Makefile

```diff
@@ -28,6 +29,11 @@ else
 		exit 22 \
 	);
 endif
+ifdef node_installed
+	node ./check-grammar.js index.html
+else
+	@echo You need node for grammer checking.
```

Comment:
One comment, though... Do we want this to not happen silently in automation if something's off with the setup, or should it fail loudly?

Reply:
That's a good point, not sure. Perhaps it's easier to continue to duplicate...

Reply:
You could check if one of https://docs.travis-ci.com/user/environment-variables/#default-environment-variables is set. Figuring out how to do that would require figuring out how Makefile syntax works though -_-.

Reply:
Similarly that could help avoid CI echoing the whole "Bikeshed is not locally installed" thing above by wrapping that in an if statement.

Reply:
@sideshowbarker I suspect this is up your alley?

Reply:
> You could check if one of https://docs.travis-ci.com/user/environment-variables/#default-environment-variables is set. Figuring out how to do that would require figuring out how Makefile syntax works though -_-.
> @sideshowbarker I suspect this is up your alley?

Yeah, I’ll take a look

Reply:
> You could check if one of https://docs.travis-ci.com/user/environment-variables/#default-environment-variables is set. Figuring out how to do that would require figuring out how Makefile syntax works though -_-.

So I see that makefile is already using some `ifdef …`. The makefile syntax for checking environment variables is to just use, e.g. `ifdef $(TRAVIS)` — that’d seem to be the simplest way to identify whether we’re being run in the Travis environment. There’s also `ifndef` for the negation of that, as well as `ifeq` and `ifneq` for any case where we’d want to actually check the value of an environment variable (rather that just checking if it’s defined.)

(I’d push a change for this to the PR branch myself, but I don’t have push perms here.)